src/onnxflow/justbuildit/export.py

```diff
@@ -0,0 +1,614 @@
+
+        input_specs = []
+        for inp, name in zip(inputs, input_names):
+            dtype = inp.dtype
+            shape = inp.shape
+            if inp.dtype == tf.float64:
+                print(f"Converting input {name} from float64 to float32")
+                dtype = tf.float32
+            if inp.dtype == tf.int64:
+                print(f"Converting input {name} from int64 to int32")
+                dtype = tf.int32
+            if inp.shape[0] is None:
+                print("Found batch size None and setting it to 1")
+                shape = (1, shape[1:])
+
+            input_specs.append(tf.TensorSpec(shape, dtype, name))
+
+        state.info.opset = build.DEFAULT_ONNX_OPSET
+
+        # Export the model to ONNX
+        tf2onnx.convert.from_keras(
+            state.model,
+            input_signature=input_specs,
+            opset=state.info.opset,
+            output_path=state.base_onnx_file,
+        )
+
+        # Save output names to ensure we are preserving the order of the outputs
+        state.expected_output_names = get_output_names(state.base_onnx_file)
+
+        state.inputs = dict(zip(tuple(input_names), tuple(inputs)))
+
+        # Save inputs and convert to fp16 and int32 (no int64 nor float32/64)
+        to_downcast = False if state.quantization_samples else True
+        tensor_helpers.save_inputs(
+            [state.inputs], state.original_inputs_file, downcast=to_downcast
+        )
+
+        # Check the if the base mode has been exported successfully
+        success_msg = "\tSuccess exporting model to ONNX"
+        fail_msg = "\tFailed exporting model to ONNX"
+        state.info.base_onnx_exported = _check_model(
+            state.base_onnx_file, success_msg, fail_msg
+        )
+
+        if state.info.base_onnx_exported:
+            state.intermediate_results = [state.base_onnx_file]
+        else:
+            msg = f"""
+            Unable to export model to ONNX using tf2onnx exporter.
+            We recommend that you modify your model until it is
+            compatible with this third party software, then re-run benchit().
+            More information may be available in the log file at **{self.logfile_path}**
+            """
+            raise exp.StageError(msg)
+
+        return state
+
+
+class OptimizeOnnxModel(stage.Stage):
+    """
+    Stage that takes an ONNX file and uses ONNX Runtime to optimize it.
+    Important because this helps to perform constant folding, Redundant
+    node eliminations, Semantics-preserving node fusions
+
+    Expected inputs:
+     - state.intermediate_results contains a single .onnx file
+
+    Outputs:
+     - A *-opt.onnx file
+    """
+
+    def __init__(self):
+        super().__init__(
+            unique_name="optimize_onnx",
+            monitor_message="Optimizing ONNX file",
+        )
+
+    def fire(self, state: build.State):
+
+        # TODO: validate this input
+        # https://git.groq.io/code/Groq/-/issues/13947
```

Comment:
* Remove link to GroqFlow issue
* Replace with MLAgility issue
* Cross reference: Mention MLAgility issue in the GroqFlow issue